src/Datasource/EntityTrait.php

```diff
@@ -281,8 +299,10 @@ public function &get($property)
 
         if ($this->_methodExists($method)) {
             $result = $this->{$method}($value);
-            return $result;
+            return $recursor($result, $nested);
```

Comment:
Also note that the final `return` operates on `$value` where the nested return above it operates on `$result`.